src/oscar/apps/offer/abstract_models.py

```diff
@@ -26,6 +26,60 @@
 BrowsableRangeManager = get_class('offer.managers', 'BrowsableRangeManager')
 
 
+@python_2_unicode_compatible
+class BaseOfferMixin(models.Model):
+    class Meta:
+        abstract = True
+
+    def proxy(self):
+        """
+        Return the proxy model
+        """
+        klassmap = self.proxy_map
+        # Short-circuit logic if current class is already a proxy class.
+        if self.__class__ in klassmap.values():
+            return self
+
+        field_dict = dict(self.__dict__)
+        for field in list(field_dict.keys()):
+            if field.startswith('_'):
+                del field_dict[field]
+
+        if self.proxy_class:
+            klass = utils.load_proxy(self.proxy_class)
+            # Short-circuit again.
+            if self.__class__ == klass:
+                return self
+            return klass(**field_dict)
+        if self.type in klassmap:
+            return klassmap[self.type](**field_dict)
+        raise RuntimeError("Unrecognised condition type (%s)" % self.type)
```

Comment:
The wording of this exception needs to change, since it no longer applies only to `Condition`s.